File: app/src/main/java/io/github/edwinvanrooij/noteboard/GameListener.java
package io.github.edwinvanrooij.noteboard;

public interface GameListener {
  void gameStarted();
  void onAccuracyChange(int accuracy);
  void onScoreChange(int newScore);
  void onNewNote(Note note);
  void onCorrectGuess(Note note);
  void onIncorrectGuess(Note note);
}

File: app/src/main/java/io/github/edwinvanrooij/noteboard/GameListener.kt
package io.github.edwinvanrooij.noteboard;

interface GameListener {
    fun onGameStart()
    fun onAccuracyChange(accuracy: Double)
    fun onScoreChange(newScore: Int)
    fun onNewNote(note: Note)
    fun onCorrectGuess(note: Note)
    fun onIncorrectGuess(guess: NoteName, correct: Note)
}
